### ISSUE TITLE
403 response code under PHP 7.0.9 instead of 404

### ISSUE THREAD

**Issue by Sulpher** [joomla/joomla-cms#11364]
#### Steps to reproduce the issue
I noticed there is no 404 response code in Joomla 3.6.0 in case of wrong URL. Joomla gives 403. Here is page header:

HTTP/1.1 403
Server: nginx
Date: Sat, 30 Jul 2016 08:15:06 GMT
Content-Type: text/html; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Keep-Alive: timeout=60
X-Powered-By: PHP/7.0.9
Expires: Wed, 17 Aug 2005 00:00:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Last-Modified: Sat, 30 Jul 2016 08:15:06 GMT
Content-Encoding: gzip

I am using PHP 7.0.9. If I switch server to PHP 5.6 - all is fine - Joomla gives 404 response code. 

#### Expected result
404 response code.

#### Actual result
403 response code. 

#### System information (as much as possible)
Database Version	5.5.47-MariaDB
Database Collation	utf8_general_ci
Database Connection Collation	utf8mb4_general_ci
PHP Version	7.0.9
Web Server	Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips mod_fcgid/2.3.9 PHP/5.4.45
WebServer to PHP Interface	cgi-fcgi
Joomla! Version	Joomla! 3.6.0 Stable [ Noether ] 12-July-2016 21:14 GMT
Joomla! Platform Version	Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
User Agent	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36

**Comment by n9iels** [comment]
I can confirm this issue. When running PHP 5.6.24 and type a random URL I get a `404 - Article not found` error.
When running PHP 7.0.9 I get a `0 - Call to a member function get() on null` error.

I tested this on latest staging with the error reporting on `Develop`. On another Joomla! 3.6 stable website running PHP 7.0.8 there is no problem. Not sure if this is a problem for the current staging or the stable version.

**Comment by wilsonge** [comment]
<img width="1022" alt="screen shot 2016-07-30 at 12 02 09" src="https://cloud.githubusercontent.com/assets/1986000/17269904/8c84d862-564d-11e6-8563-3e20d31fa136.png">

Georges-MacBook-Pro:jed3-dev george$ curl http://localhost/~george/joomla-cms/?option=com_fooooo -I
HTTP/1.1 404 Component not found.
Date: Sat, 30 Jul 2016 11:03:33 GMT
Server: Apache/2.4.18 (Unix) PHP/7.0.9
X-Powered-By: PHP/7.0.9
Set-Cookie: 145fa16e3cae7251293782cff47cfb8d=c8et1ipfanhgt7bp67d9snc016; path=/; HttpOnly
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/html; charset=UTF-8

I can't confirm such an issue on staging with PHP 7.0.9 :(

**Comment by brianteeman** [comment]
I see the original report was on nginx and not Apache. Perhaps that's the
issue?

**Comment by brianteeman** [comment]
I can confirm multiple issues with Joomla 3.6.0 and 3.6.1rc1 on php7.0.9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11364">issues.joomla.org/joomla-cms/11364</a>.</sub>

**Comment by wilsonge** [comment]
Maybe. Generating a 404 a different way (how I got a 403 on your test site @brianteeman ) I got on this local setup

```
Georges-MacBook-Pro:joomla-cms george$ curl -I http://localhost/~george/joomla-cms/foobar
HTTP/1.1 500 Call to a member function get() on null
Date: Sat, 30 Jul 2016 11:41:25 GMT
Server: Apache/2.4.18 (Unix) PHP/7.0.9
X-Powered-By: PHP/7.0.9
Set-Cookie: 145fa16e3cae7251293782cff47cfb8d=t97887dhgilni62861em6f0ig6; path=/; HttpOnly
Cache-Control: no-cache
Pragma: no-cache
Connection: close
Content-Type: text/html; charset=UTF-8
```

**Comment by n9iels** [comment]
@wilsonge your result is correct, but when you type someting like `http://joomla-cms.dev/index.php/test` I get an `0 - Call to a member function get() on null` error

**Comment by brianteeman** [comment]
I am closing this as it looks like it is a duplicate of #11360 which has a PR for testing<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11364">issues.joomla.org/joomla-cms/11364</a>.</sub>

**Comment by n9iels** [comment]
@brianteeman the issue I have seems like a different issue?

**Comment by brianteeman** [comment]
Re-opened - sorry it is not the exact same issue<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11364">issues.joomla.org/joomla-cms/11364</a>.</sub>

**Comment by brianteeman** [comment]
i think you are wrong and it is the same issue resolved by #11360 
http://dev-3-6-1-site-2.brian.deevop.site/ggjgjhgj with patch from #11360 applied all good

http://dev-3-6-0-site-2.brian.deevop.site/ggjgjhgj without patch applied error and 403<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11364">issues.joomla.org/joomla-cms/11364</a>.</sub>

**Comment by n9iels** [comment]
Yes you're right, when applying that PR the `0 - Call to a member function get() on null` error is gone :)

**Comment by wilsonge** [comment]
This is fixed with me merging #11360
